appinventor/components/src/com/google/appinventor/components/annotations/androidmanifest/FeatureElement.java

```diff
@@ -0,0 +1,54 @@
+// -*- mode: java; c-basic-offset: 2; -*-
+// Copyright 2025 MIT, All rights reserved
+// Released under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package com.google.appinventor.components.annotations.androidmanifest;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation to describe an <uses-feature> element required by a component
+ * so that it can be added to AndroidManifest.xml.
+ *
+ * @author https://github.com/jewelshkjony (Jewel Shikder Jony)
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+public @interface FeatureElement {
+
+  /**
+   * Specifies a single hardware or software feature used by the application as a descriptor string.
+   * Valid attribute values are listed in the
+   * <a href="https://developer.android.com/guide/topics/manifest/uses-feature-element#hw-features">Hardware features</a>
+   * and <a href="https://developer.android.com/guide/topics/manifest/uses-feature-element#sw-features">Software features</a> sections.
+   * These attribute values are case-sensitive.
+   *
+   * @return the name of the element
+   */
+  String name();
+
+  /**
+   * Boolean value that indicates whether the application requires the featureContents of the xml file.
+   * Declaring true for a feature indicates that the application can't function,
+   * or isn't designed to function, when the specified feature isn't present on the device.
+   * Declaring false for a feature indicates that the application uses the feature if present on the device,
+   * but that it is designed to function without the specified feature if necessary.
+   *
+   * @return Returns true if required otherwise false
+   */
+  boolean required() default true;
```

Comment:
I think that it would be better to have the default here be false, with the reasoning being that if a component/extension developer needs something to be necessary they should declare that explicitly.